src/types/closure.md

```diff
@@ -178,3 +288,117 @@ Because captures are often by reference, the following general rules arise:
 [`Sync`]: ../special-types-and-traits.md#sync
 [closure expression]: ../expressions/closure-expr.md
 [derived]: ../attributes/derive.md
+
+## Drop Order
+
+If a closure captures a field of a composite types such as structs, tuples, and enums by ByValue, the field's lifetime would now be tied to the closure. As a result, it is possible for disjoint fields of a composite types to be dropped at different times.
+
+```rust
+{
+    let tuple = 
+      (String::from("foo"), String::from("bar")); // --+
+    { //                                               |
+        let c = || { // ----------------------------+  |
+            // tuple.0 is captured into the closure |  |
+            drop(tuple.0); //                 |  |
+        }; //                                       |  |
+    } // 'c' and 'tuple.0' dropped here ------------+  |
+} // tuple.1 dropped here -----------------------------+
+```
+
+## Wild Card Patterns
+Closures now only capture data that needs to be read, which means the following closures will not capture `x`
+
+```rust
+let x = 10;
+let c = || {
+    let _ = x;
+};
+
+let c = || match x {
+    _ => println!("Hello World!")
+};
+```
+
+
+# Edition 2018 and before
+
+## Closure types difference
+
+In Edition 2018 and before, a closure would capture variables in its entirety. This meant that for the example used in the [Closure types]() section the generated closure type would look something like this:
```

Comment:
```suggestion
In Edition 2018 and before, a closure would capture variables in its entirety. This means that for the example used in the [Closure types] section, the generated closure type would instead look something like this:
```

Reply:
we should create a link above and link them here? I think if look at a rust RFC there should be expamples for it.

I'm refering to the fact that [Clsoure types] doesn't have a link associated with it